src/main/java/frc/robot/commands/PhotonMoveToTarget.java

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+//imports
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.subsystems.SwerveSubsystem;
+import frc.robot.subsystems.PhotonSubsystem;
+
+//class
+public class PhotonMoveToTarget extends Command {
+  //declerations
+  Translation2d targetOffset;
+  boolean centerTarget;
+  Rotation2d desiredHeading;
+  double tolerance=0.01;
+
+
+  public PhotonMoveToTarget(Translation2d _targetOffset, double _tolerance) {
+    addRequirements(SwerveSubsystem.getInstance());
+    addRequirements(PhotonSubsystem.getInstance());
+    targetOffset = _targetOffset;
+    centerTarget=true;
+    tolerance=_tolerance;
+  }
+
+  //if you want a direction at the stopping point
+  public PhotonMoveToTarget(Translation2d _targetOffset, Rotation2d _desiredHeading, double _tolerance) {
+    addRequirements(SwerveSubsystem.getInstance());
+    addRequirements(PhotonSubsystem.getInstance());
+    targetOffset = _targetOffset;
+    desiredHeading = _desiredHeading;
+    centerTarget=false;
+    tolerance=_tolerance;
+  }
+
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    SwerveSubsystem.getInstance().resetDriveToPose();
+  }
+
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    Translation2d setPoint = PhotonSubsystem.getInstance().getTargetPos();
+    Rotation2d rotationSetPoint = PhotonSubsystem.getInstance().getTargetRotation();
+    if (centerTarget){
+      SwerveSubsystem.getInstance().driveToPose(new Pose2d(setPoint.plus(targetOffset), rotationSetPoint));
+    }else{
+      SwerveSubsystem.getInstance().driveToPose(new Pose2d(setPoint.plus(targetOffset), desiredHeading));
+    }
+  }
+ 
+
+  // Called once the command ends or is interrupted.
+  @Override
+  public void end(boolean interrupted) {
+    // SwerveSubsystem.getInstance().stopMotors();
+  }
+
+  
+  // Returns true when the command should end.
+  @Override
+  public boolean isFinished() {
+    return SwerveSubsystem.getInstance().isAtPose(tolerance, tolerance*5);
```

Comment:
This is still not ideal. Making the angular tolerance in radians be 5 times to tolerance in meters is weird.

I think the best way to go about this is to change the `double _tolerance `in the constructor is be `boolean isWaypoint`. Then you can do an if statement here for if it's a waypoint and grab values from PhotonConfig for the tolerance. Here's my guess for numbers that will probably just work:
```Java
    public static final double POS_TOLERANCE = 0.05; // meters
    public static final double ANGLE_TOLERANCE = Math.toRadians(1.0);
    public static final double WAYPOINT_POS_TOLERANCE = 0.3; // meters
    public static final double WAYPOINT_ANGLE_TOLERANCE = Math.toRadians(4.0);
```

Reply:
We might also want to make sure the chassis isn't moving for the final pose it goes to (non waypoint commands). It could be possible for vision to take control a little late and result in the chassis reaching the correct position but with too much velocity that is overshoots the position. You can use a method in SwerveSubsystem to check called isChassisMoving().